### PR TITLE
WIP - added swagger documentation to news-flash and infographics-data

### PR DESCRIPTION
**Pull request by ziv17**
This is still WIP, just to get feedback on the documentation. Currently the code mostly does not use the swagger definitions, they only contribute the documentation (reachable via `/` url)
Added Swagger documentation to api/news-flash-new, api/news-flash/<id>, and api/infographics-data
Regarding `/api/news-flash`. I think Swagger does not support marshaling of a list on the outer level (only pairs of fields and values). So I put the documentation on `...-new`.
Please comment.
Should I continue to add URLs, or move to actually use Swagger definition to retrieve input parameters for above URLs?

### CODE REVIEW

File: anyway/flask_app.py [hunk 1]
```diff
@@ -1236,9 +1303,27 @@ def oauth_callback(provider: str) -> Response:
...
     Returns infographics-data API
 """
+parser = reqparse.RequestParser()
+parser.add_argument('id', type=int, help='News flash id')
+parser.add_argument('years_ago', type=int, default=5, help='Number of years back to consider accidents')
+
+resp_mod = api.model('infographics_data', {
+    'rank': fields.Raw(readonly=True, description='The task unique identifier'),
+    'task': fields.Raw(required=True, description='The task details')
+})
```
Comment (BarVolunteering) [inline anyway/flask_app.py:1313]: Is this code describe the JSON returning from "/api/infographics-data" API? - I think some of the fields are missing

File: anyway/flask_app.py [hunk 2]
```diff
@@ -1135,15 +1139,78 @@ def load_identity_when_session_expires():
...
+
+@api.route("/api/news-flash-new", methods=["GET"])
+class RetrieveNewsFlash(Resource):
+
+    @api.doc('get news flash records')
+    @api.expect(parser)
+    @api.response(404, 'Parameter value not supported or missing')
+    @api.response(200, 'Retrieve news-flash items filtered by given parameters', news_flash_list_model)
+    def get(self):
+        return news_flash()
```
Comment (BarVolunteering) [inline anyway/flask_app.py:1211]: I am not sure that i understand the concept of `@api.expect`, in the case of `RetrieveSingleNewsFlash` the `single_news_flash` function is getting params but `@api.expect(parser)` is not used and in the case of `RetrieveNewsFlash` the `news_flash` function is not getting any params but `@api.expect(parser)` is used.

Why is that?

File: anyway/flask_app.py [hunk 3]
```diff
@@ -1135,15 +1139,78 @@ def load_identity_when_session_expires():
...
-    endpoint=None,
-    view_func=single_news_flash,
-    methods=["GET"],
-)
+# app.add_url_rule(
+#     "/api/news-flash/<int:news_flash_id>",
+#     endpoint=None,
+#     view_func=single_news_flash,
+#     methods=["GET"],
+# )
```
Comment (BarVolunteering) [inline anyway/flask_app.py:1147]: Why not delete it?

**Review by BarVolunteering — COMMENTED** [review]
Good job!

**Review by BarVolunteering — APPROVED** [review]
Good work!
Few comment above.

File: anyway/flask_app.py (hunk 1 above)
Comment (atalyaalon) [reply anyway/flask_app.py:1313]: Not sure I understand this part of resp_mod - should it represent infographics_data? What is task?

**Review by atalyaalon — CHANGES_REQUESTED** [review]
Great work, I added one question.
I think we should move to use Swagger definition to retrieve input parameters for above URLs
@yuvalbl any thoughts?

File: anyway/flask_app.py (hunk 1 above)
Comment (ziv17) [reply anyway/flask_app.py:1313]: This was an initial trial I forgot to remove. Removing it until I will document this API. I leave only documentation of input parameters for `infographics-data` API.

File: anyway/flask_app.py (hunk 2 above)
Comment (ziv17) [reply anyway/flask_app.py:1211]: You are right. In the single news flash, the id is part of the URL, so it is not described in the `@api.exect` construct. In the `RetrieveNesFlash` the parameters are described in the `api.expect`. In the current implementation they are not passed to the `news_flash()` method, as it is a partial implementation. I will complete it it, and then they will be passed to the `news_flash()` method, and will be consumed directly from the parser (rather then from the `request`).
`
@api.route("/api/news-flash-new", methods=["GET"])
class RetrieveNewsFlash(Resource):

    @api.doc('get news flash records')
    @api.expect(parser)
    @api.response(404, 'Parameter value not supported or missing')
    @api.response(200, 'Retrieve news-flash items filtered by given parameters', news_flash_list_model)
    def get(self):
        args = parser.parse_args()
        return news_flash_new(args)

`

**Pull request merged**